Packs/CiscoEmailSecurity/Integrations/CiscoEmailSecurity/CiscoEmailSecurity.py

```diff
@@ -0,0 +1,629 @@
+                                    sender=message.get('sender'), recipient=message.get('recipient'),
+                                    subject=message.get('subject'), timestamp=message.get('timestamp'))
+    headers = ['message_id', 'direction', 'sender', 'recipient', 'subject', 'timestamp']
+    human_readable = tableToMarkdown('CiscoEmailSecurity Messages', readable_output, headers, removeNull=True)
+    return human_readable
+
+
+def response_data_to_context_and_human_readable(response_data):
+    context_data = response_data.get('data')
+    context_message = context_data.get('messages')
+    context_message['mid'] = context_message.get('mid', [None])[0]
+    human_readable = details_get_to_human_readable(context_message)
+    return context_message, human_readable
+
+
+def list_get_message_details_command(client, args):
+    url_suffix_to_filter_by = build_url_params_for_get_details(args, '/sma/api/v2.0/message-tracking/details')
+    message_get_details_response_data = client.list_get_message_details(url_suffix_to_filter_by)
+    message_data, human_readable = response_data_to_context_and_human_readable(message_get_details_response_data)
+    return CommandResults(
+        readable_output=human_readable,
+        outputs_prefix='CiscoEmailSecurity.Message',
+        outputs_key_field='messages.mid',
+        outputs=message_data
+    )
+
+
+def list_get_dlp_details_command(client, args):
+    url_suffix_to_filter_by = build_url_params_for_get_details(args, '/sma/api/v2.0/message-tracking/dlp-details')
+    message_get_details_response_data = client.list_get_dlp_details(url_suffix_to_filter_by)
+    message_data, human_readable = response_data_to_context_and_human_readable(message_get_details_response_data)
+    return CommandResults(
+        readable_output=human_readable,
+        outputs_prefix='CiscoEmailSecurity.Dlp',
+        outputs_key_field='messages.mid',
+        outputs=message_data
+    )
+
+
+def list_get_amp_details_command(client, args):
+    url_suffix_to_filter_by = build_url_params_for_get_details(args, '/sma/api/v2.0/message-tracking/amp-details')
+    message_get_details_response_data = client.list_get_amp_details(url_suffix_to_filter_by)
+    message_data, human_readable = response_data_to_context_and_human_readable(message_get_details_response_data)
+    return CommandResults(
+        readable_output=human_readable,
+        outputs_prefix='CiscoEmailSecurity.Amp',
+        outputs_key_field='messages.mid',
+        outputs=message_data
+    )
+
+
+def list_get_url_details_command(client, args):
+    url_suffix_to_filter_by = build_url_params_for_get_details(args, '/sma/api/v2.0/message-tracking/url-details')
+    message_get_details_response_data = client.list_get_url_details(url_suffix_to_filter_by)
+    message_data, human_readable = response_data_to_context_and_human_readable(message_get_details_response_data)
+    return CommandResults(
```

Comment:
Shouldn't you create URL indicator context as well?

Reply:
@ChanochShayner what about this one?

Reply:
@ChanochShayner any update regarding the URL indicator?

Reply:
The fields in response data that returns do not match the fields in the URL context standard,
I asked @Arsenikr to check this.

Reply:
According to @Arsenikr - There is no need to do this.
We align according to this standard only if we make a command !Url and treat it as an indicator.